### PR TITLE
Fix of #40 - checking KDocs for special methods

### LINKED ISSUE #40: [Formatting rules] Improve analysis of KDocs for functions
In addition to rule 2.5:
* [x] Do not force KDocs for methods with name "(get|set).*" and single line of body
* [x] Do not force `@return`, `@param` tag for getters/setters at all.
* [ ] The following KDocs are considered useless: if they start with words return/get/set and contains one or two more words
* [x] Do not force KDoc for standard methods like `toString`, `equals` and `hashCode`

### PR DESCRIPTION
**Pull request by petertrr**
Closes #40

### What's done:
* Implemented logic for checking KDocs for setters/getters
* Implemented logic for checking KDocs for equals, hashCode, toString
* Implemented logic for checking if KDoc is trivial
* Added tests

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/KdocMethods.kt [hunk 1]
```diff
@@ -55,9 +57,14 @@ import org.jetbrains.kotlin.kdoc.psi.impl.KDocTag
...
-    // expression body of function can have a lot of 'ElementType's, this list might be not full
-    private val expressionBodyTypes = setOf(BINARY_EXPRESSION, CALL_EXPRESSION, LAMBDA_EXPRESSION, REFERENCE_EXPRESSION,
-            CALLABLE_REFERENCE_EXPRESSION, SAFE_ACCESS_EXPRESSION, WHEN_CONDITION_WITH_EXPRESSION, COLLECTION_LITERAL_EXPRESSION)
+    companion object {
+        // expression body of function can have a lot of 'ElementType's, this list might be not full
+        private val expressionBodyTypes = setOf(BINARY_EXPRESSION, CALL_EXPRESSION, LAMBDA_EXPRESSION, REFERENCE_EXPRESSION,
+                CALLABLE_REFERENCE_EXPRESSION, SAFE_ACCESS_EXPRESSION, WHEN_CONDITION_WITH_EXPRESSION, COLLECTION_LITERAL_EXPRESSION)
+
+        // these methods do not need mandatory documentation
+        private val standardMethods = listOf("equals", "hashCode", "toString", "clone", "finalize")
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/KdocMethods.kt:66]: how about c o n f i g u r a t i o n?

Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/KdocMethods.kt:66]: at least put FixMe

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/KdocMethods.kt:66]: You mean, someone could possibly want to make documentation of `equals` mandatory, or that user should be able to add their own methods which do not need to be documented? In the latter case, we could have two separate configuration options: `standard methods` and `custom methods`.

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/KdocMethods.kt:69]: Eventually decided not to allow customizations here.

**Pull request merged**